docs/kb/semgrep-ci/github-repository-rulesets-semgrep.md

```diff
@@ -0,0 +1,75 @@
+---
+tags:
+  - GitHub
+  - Repository rulesets
+description: Setting up GitHub required workflows for Semgrep scans.
+---
+
+# Using GitHub repository rulesets to implement Semgrep
+
+If your GitHub organization has hundreds or thousands of repositories, setting up Semgrep by adding the `.github/workflows/semgrep.yml` file to all repositories can be a time-consuming task. You can instead use [repository rulesets](https://docs.github.com/en/enterprise-cloud@latest/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/creating-rulesets-for-a-repository#introduction) to add Semgrep as a workflow that is [required to pass before merging](https://docs.github.com/en/enterprise-cloud@latest/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/available-rules-for-rulesets#require-workflows-to-pass-before-merging). 
+
+This allows you to set up a centralized repository with a single Semgrep workflow file that can be applied to as many repositories as desired. Formerly, this feature was called [required workflows](https://github.blog/changelog/2023-08-02-github-actions-required-workflows-will-move-to-repository-rules/).
+
+## Setting up the central Semgrep scan workflow
+
+To use the Semgrep workflow in other repositories, you can either create a new repository with the Semgrep workflow file, or add it to an existing repository where you store common workflows. 
+
+When a workflow is required to pass before merging, the workflow file needs to be in a repository that matches the widest visibility of the repositories you want to run it in. For example, if you want to run Semgrep on public, internal, and private repositories, the repository containing the workflow file must be public.
```

Comment:
Just curious - what if the workflow is not required to pass?

Reply:
The only option for repository rulesets & workflows is to require them to pass before merging. If you want to do something else you'd have to use some other mechanism.

Reply:
In that case I think we can say "the workflow fiiles neds to be run..." without the prior clause, because there's no other case.